--- /home/mahaloz/kwt/elfmain/.scratch/realsweep/off/O0_dash_dash.c	2026-09-16 20:49:56.587047988 +0000
+++ /home/mahaloz/kwt/elfmain/.scratch/realsweep/on/O0_dash_dash.c	2026-09-16 20:49:52.493039951 +0000
@@ -657,7 +657,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_23fc0)(sub_fd15,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_23fc0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
@@ -6345,8 +6345,8 @@
   sub_8b10(a0,a1,1);
 }
 
-// Function: sub_fd15 @ 0xfd15
-void sub_fd15(unsigned int a0,unsigned long a1)
+// Function: main @ 0xfd15
+int main(int argc,char **argv,char **envp)
 {
   char *v1; // rax
   char v2 [24];
@@ -6360,7 +6360,7 @@
     dat_266e0 = getpid();
     sub_1a81e();
     sub_105e6(v2);
-    if (sub_119d6(a0,a1)) {
+    if (sub_119d6(argc,argv)) {
       sub_100c0("/etc/profile");
       sub_100c0("$HOME/.profile");
     }
