--- /home/mahaloz/kwt/elfmain/.scratch/realsweep/off/O0_grep_grep.c	2026-09-16 20:46:08.717594852 +0000
+++ /home/mahaloz/kwt/elfmain/.scratch/realsweep/on/O0_grep_grep.c	2026-09-16 20:46:02.177581644 +0000
@@ -1017,7 +1017,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_32fd0)(sub_a7d1,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_32fd0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
@@ -3327,88 +3327,84 @@
   return v5;
 }
 
-// Function: sub_a7d1 @ 0xa7d1
-unsigned char sub_a7d1(int a0,long a1) // early-return, ternary x2, warn: iteboolean: re-rolled 2 0/1 select diamond(s) to a boolean assignment in sub_a7d1
+// Function: main @ 0xa7d1
+int main(int argc,char **argv,char **envp) // ternary x3, warn: iteboolean: re-rolled 2 0/1 select diamond(s) to a boolean assignment in main
 {
-  int v1;
-  unsigned long v10;
-  unsigned long v11;
-  unsigned char v12; // al
-  unsigned int v13; // eax
-  char *v14; // rax
-  long v15;
-  unsigned int v16; // ebx
-  long v17; // stack - 0x1c8
-  long v18; // stack - 0x1d0
-  long v19; // stack - 0x148
-  long *v2;
-  long v20; // stack - 0xb8
-  long v21; // stack - 0x1d8
-  char v22; // stack - 0x22
-  char v23 [8];
-  bool v24; // stack - 0x202
-  int v25; // stack - 0x200
-  int v26; // stack - 0x1fc
-  unsigned int v27; // stack - 0x1f8
-  int v28; // stack - 0x1f4
-  int v29; // stack - 0x1f0
-  unsigned long *v3;
-  int v30; // stack - 0x1ec
-  int v31; // stack - 0x1e8
-  char *v32; // stack - 0x1b8
-  void *v33; // stack - 0x1b0
-  long v34; // stack - 0x1a8
-  char *v35; // stack - 0x1a0
-  long *v36; // stack - 0x198
-  char *v37; // stack - 0x190
-  unsigned long v38; // stack - 0x188
-  char *v39; // stack - 0x180
-  char v4 [16];
-  unsigned long v40; // stack - 0x178
-  long v41; // stack - 0x170
-  long v42; // stack - 0x168
-  unsigned long v43; // stack - 0x160
-  long v44; // stack - 0x158
-  char *v45; // stack - 0x150
-  long v46; // stack - 0x140
-  unsigned long v47; // stack - 0x138
-  unsigned int v48; // stack - 0x130
-  unsigned int v49; // stack - 0x12c
-  char v5 [16];
-  unsigned long v50; // stack - 0x128
-  unsigned long v51; // stack - 0x120
-  unsigned long v52; // stack - 0x118
-  unsigned long v53; // stack - 0x110
-  unsigned long v54; // stack - 0x108
-  unsigned long v55; // stack - 0x100
-  unsigned long v56; // stack - 0xf8
-  unsigned long v57; // stack - 0xf0
-  unsigned long v58; // stack - 0xe8
-  unsigned long v59; // stack - 0xe0
-  bool v6;
-  unsigned long v60; // stack - 0xd8
-  unsigned long v61; // stack - 0xd0
-  unsigned long v62; // stack - 0xc8
-  unsigned long v63; // stack - 0xc0
-  long v64; // stack - 0xb0
-  char v65; // stack - 0x23
-  char v66; // stack - 0x21
-  char *v7;
-  unsigned int v8;
+  char **v1;
+  int v10; // eax
+  char *v11;
+  long v12;
+  long v13; // stack - 0x1c8
+  long v14; // stack - 0x1d0
+  long v15; // stack - 0x148
+  long v16; // stack - 0xb8
+  long v17; // stack - 0x1d8
+  char v18; // stack - 0x22
+  char v19 [8];
+  char v2 [16];
+  bool v20; // stack - 0x202
+  int v21; // stack - 0x200
+  int v22; // stack - 0x1fc
+  unsigned int v23; // stack - 0x1f8
+  int v24; // stack - 0x1f4
+  int v25; // stack - 0x1f0
+  int v26; // stack - 0x1ec
+  int v27; // stack - 0x1e8
+  char *v28; // stack - 0x1b8
+  void *v29; // stack - 0x1b0
+  char v3 [16];
+  long v30; // stack - 0x1a8
+  char *v31; // stack - 0x1a0
+  char **v32; // stack - 0x198
+  char *v33; // stack - 0x190
+  unsigned long v34; // stack - 0x188
+  char *v35; // stack - 0x180
+  unsigned long v36; // stack - 0x178
+  long v37; // stack - 0x170
+  long v38; // stack - 0x168
+  unsigned long v39; // stack - 0x160
+  bool v4;
+  long v40; // stack - 0x158
+  char *v41; // stack - 0x150
+  long v42; // stack - 0x140
+  unsigned long v43; // stack - 0x138
+  unsigned int v44; // stack - 0x130
+  unsigned int v45; // stack - 0x12c
+  unsigned long v46; // stack - 0x128
+  unsigned long v47; // stack - 0x120
+  unsigned long v48; // stack - 0x118
+  unsigned long v49; // stack - 0x110
+  unsigned char v5; // al
+  unsigned long v50; // stack - 0x108
+  unsigned long v51; // stack - 0x100
+  unsigned long v52; // stack - 0xf8
+  unsigned long v53; // stack - 0xf0
+  unsigned long v54; // stack - 0xe8
+  unsigned long v55; // stack - 0xe0
+  unsigned long v56; // stack - 0xd8
+  unsigned long v57; // stack - 0xd0
+  unsigned long v58; // stack - 0xc8
+  unsigned long v59; // stack - 0xc0
+  char v6;
+  long v60; // stack - 0xb0
+  char v61; // stack - 0x23
+  char v62; // stack - 0x21
+  unsigned int v7; // eax
+  unsigned int v8; // eax
   int v9;
   
-  v32 = NULL;
-  v21 = 0;
-  v18 = 0;
-  v25 = -1;
+  v28 = NULL;
+  v17 = 0;
+  v14 = 0;
+  v21 = -1;
   dat_33080 = 2;
-  v27 = 0;
+  v23 = 0;
   dat_33183 = 10;
   dat_3384c = 0xffffffff;
   dat_33870 = 0x7fffffffffffffff;
   dat_33858 = -1;
   dat_33860 = -1;
-  v17 = -1;
+  v13 = -1;
   dat_3375c = '\0';
   setlocale(6,(char *)0x290d2);
   bindtextdomain("grep","/usr/local/share/locale");
@@ -3416,94 +3412,76 @@
   sub_1bf18(0x331a0);
   sub_285f0(sub_6f2c);
   sub_112dd(0);
-  v26 = 0;
-  v14 = (long)sub_208d6(0,0,sub_6760,sub_67f8,0);
-  dat_33790 = (long)v14;
-  if (!v14)
+  v22 = 0;
+  dat_33790 = sub_208d6(0,0,sub_6760,sub_67f8,0);
+  if (!dat_33790)
     sub_27122(); // no-return
 label_b57f:
   do {
-    v1 = optind;
-    v13 = sub_9c9f(a0,a1,&v17);
-    v9 = v13;
-    if (v13 == -1) {
+    v9 = optind;
+    v10 = sub_9c9f(argc,argv,&v13);
+    if (v10 == -1) {
       if (dat_33754) {
-        v14 = (unsigned long)sub_1ff0c();
-        sub_2690a(stdout,v14,"GNU grep","3.8",0);
-        v14 = gettext("Written by Mike Haertel and others; see\n<https://git.sv.gnu.org/cgit/grep.git/tree/AUTHORS>.");
-        puts(v14);
-        v12 = 0;
-        return v12;
+        sub_2690a(stdout,sub_1ff0c(),"GNU grep","3.8",0);
+        puts(gettext("Written by Mike Haertel and others; see\n<https://git.sv.gnu.org/cgit/grep.git/tree/AUTHORS>."));
+        return 0;
       }
       if (dat_33750)
         sub_99b6(0); // no-return
-      if (v32) { // branch-flip
-        if (!v21) {
+      if (v28) { // branch-flip
+        if (!v17) {
           dat_33851 = dat_33851 != '\x01';
           dat_33181 = '\0';
-          v12 = '\0';
-          v21 = 1;
-          *v32 = '\n';
-          dat_33182 = v12;
+          dat_33182 = '\0';
+          v17 = 1;
+          *v28 = '\n';
         }
       }
       else {
-        if (a0 <= optind)
+        if (argc <= optind)
           sub_99b6(2); // no-return
-        v1 = optind + 1;
-        v14 = (long)(long)optind;
-        v14 = (unsigned long *)(a1 + (long)v14 * 8);
-        v37 = *(char **)v14;
-        v12 = (((v25 != 2) && (*v37 == '\\')) && (v14 = &v37[1], *v14 == '-'));
-        v14 = &v37[v12];
-        optind = v1;
-        v14 = (char *)sub_270f1(v14);
-        v32 = v14;
-        dat_33780 = v14;
-        v14 = (unsigned long)strlen(v14);
-        v11 = (unsigned long)v14;
-        v38 = (unsigned long)v14;
-        v14 = &v32[(long)v14];
-        *v14 = '\n';
-        v14 = (long)sub_687b(v32,0,v11 + 1,0x290d2);
-        v21 = (long)v14;
+        v9 = optind + 1;
+        v33 = argv[optind];
+        v5 = (((v21 != 2) && (*v33 == '\\')) && (v33[1] == '-'));
+        optind = v9;
+        dat_33780 = (char *)sub_270f1(&v33[v5]);
+        v28 = dat_33780;
+        v34 = strlen(dat_33780);
+        v28[v34] = '\n';
+        v17 = sub_687b(v28,0,v34 + 1,0x290d2);
       }
-      v14 = (long)(v21 + -1);
-      v21 = (long)v14;
+      v17 -= 1;
       sub_20b56(dat_33790);
-      v6 = 0;
-      if ((dat_338b1 != '\x01') && (v13 = fstat(1,&v19), !v13)) {
-        v13 = v48 & 0xf000;
-        if (v13 != 0x8000) { // branch-flip
-          v13 = v48 & 0xf000;
-          if (v13 == 0x2000) {
-            v13 = stat("/dev/null",&v20);
-            if (((v13) || (v46 != v64)) || (v19 != v20))
-              v6 = 1;
+      v4 = 0;
+      if ((dat_338b1 != '\x01') && (!fstat(1,&v15))) {
+        if ((v44 & 0xf000) != 0x8000) { // branch-flip
+          if ((v44 & 0xf000) == 0x2000) {
+            if (((stat("/dev/null",&v16)) || (v42 != v60)) || (v15 != v16))
+              v4 = 1;
             else {
               dat_338b2 = '\x01';
             }
           }
         }
         else {
-          dat_336c0 = v19;
-          dat_336c8 = v46;
-          dat_336d8 = CONCAT44(v49,v48);
-          dat_336d0 = v47;
-          dat_336e0 = v50;
-          dat_336e8 = v51;
-          dat_336f0 = v52;
-          dat_336f8 = v53;
-          dat_33700 = v54;
-          dat_33708 = v55;
-          dat_33710 = v56;
-          dat_33718 = v57;
-          dat_33720 = v58;
-          dat_33728 = v59;
-          dat_33730 = v60;
-          dat_33738 = v61;
-          dat_33740 = v62;
-          dat_33748 = v63;
+          dat_336c0 = v15;
+          dat_336c8 = v42;
+          dat_336d8 = CONCAT44(v45,v44);
+          dat_336d0 = v43;
+          dat_336e0 = v46;
+          dat_336e8 = v47;
+          dat_336f0 = v48;
+          dat_336f8 = v49;
+          dat_33700 = v50;
+          dat_33708 = v51;
+          dat_33710 = v52;
+          dat_33718 = v53;
+          dat_33720 = v54;
+          dat_33728 = v55;
+          dat_33730 = v56;
+          dat_33738 = v57;
+          dat_33740 = v58;
+          dat_33748 = v59;
         }
       }
       if (dat_338b2 || dat_338b1) { // branch-flip
@@ -3515,26 +3493,22 @@
       else if (dat_33844) goto label_b9b9;
       dat_33850 = dat_338b0 || dat_33868;
       if (dat_33860 <= -1)
-        dat_33860 = v17;
+        dat_33860 = v13;
       if (dat_33858 <= -1)
-        dat_33858 = v17;
-      if (((!dat_33870) || ((((!v21 && (dat_33851)) && (dat_33182 != '\x01')) && (dat_33181 != '\x01')))) && (dat_33844 != 2)) {
-        v12 = 1;
-        return v12;
-      }
+        dat_33858 = v13;
+      if (((!dat_33870) || ((((!v17 && (dat_33851)) && (dat_33182 != '\x01')) && (dat_33181 != '\x01')))) && (dat_33844 != 2))
+        return 1;
       if (dat_33758 == 2) {
-        v13 = !(((!v6) || (v13 = sub_2768b(), !v13)) || (v13 = isatty(1), !v13));
-        dat_33758 = v13;
+        dat_33758 = !(((!v4) || (!sub_2768b())) || (!isatty(1)));
       }
       sub_276d9();
       if (!dat_33758) goto label_bbb8;
-      v14 = getenv("GREP_COLOR");
-      v39 = v14;
-      if ((!v14) || (!*v14)) goto label_bb59;
-      v35 = v14;
+      v35 = getenv("GREP_COLOR");
+      if ((!v35) || (!*v35)) goto label_bb59;
+      v31 = v35;
       goto label_bb33;
     }
-    switch(v13) {
+    switch(v10) {
       case 0:
       case 0x55:
         goto label_b57f;
@@ -3547,36 +3521,32 @@
         sub_728f(optarg,0x33858);
         goto label_b57f;
       case 0x43:
-        sub_728f(optarg,&v17);
+        sub_728f(optarg,&v13);
         goto label_b57f;
       case 0x44:
-        v13 = strcmp(optarg,"read");
-        if (!v13) {
+        v9 = strcmp(optarg,"read");
+        if (!v9) {
           dat_337cc = 1;
           goto label_b57f;
         }
-        v13 = strcmp(optarg,"skip");
-        if (!v13) {
+        v9 = strcmp(optarg,"skip");
+        if (!v9) {
           dat_337cc = 2;
           goto label_b57f;
         }
-        v14 = gettext("unknown devices method");
-        error(2,0,v14);
+        error(2,0,gettext("unknown devices method"));
         goto label_aa4b;
       case 0x45:
-        v13 = sub_9be6("egrep",v25);
-        v25 = v13;
+        v21 = sub_9be6("egrep",v21);
         goto label_b57f;
       case 0x46:
-        v13 = sub_9be6("fgrep",v25);
-        v25 = v13;
+        v21 = sub_9be6("fgrep",v21);
         break;
       case 0x47:
-        v13 = sub_9be6("grep",v25);
-        v25 = v13;
+        v21 = sub_9be6("grep",v21);
         break;
       case 0x48:
-        v27 = 1;
+        v23 = 1;
         break;
       case 0x49:
         dat_33840 = 2;
@@ -3585,8 +3555,7 @@
         dat_33844 = 2;
         break;
       case 0x50:
-        v13 = sub_9be6("perl",v25);
-        v25 = v13;
+        v21 = sub_9be6("perl",v21);
         break;
       case 0x52:
         dat_33014 = 0x30a;
@@ -3602,8 +3571,7 @@
         dat_33754 = '\x01';
         break;
       case 0x58:
-        v13 = sub_9be6(optarg,v25);
-        v25 = v13;
+        v21 = sub_9be6(optarg,v21);
         break;
       case 0x5a:
         dat_3384c = 0;
@@ -3618,92 +3586,71 @@
         dat_33868 = '\x01';
         break;
       case 100:
-        v14 = (long)sub_10c86("--directories",optarg,0x32700,0x293a8,4,dat_33078,1);
-        dat_33010 = *(int *)((long)v14 * 4 + 0x293a8);
+        v12 = sub_10c86("--directories",optarg,0x32700,0x293a8,4,dat_33078,1);
+        dat_33010 = *(int *)(v12 * 4 + 0x293a8);
         if (dat_33010 == 3)
-          v26 = v1;
+          v22 = v9;
         break;
       case 0x65:
-        v14 = (unsigned long)strlen(optarg);
-        v43 = (unsigned long)v14;
-        v14 = (long)(v21 - v18);
-        v14 = (long)(v43 + (long)v14);
-        v14 = (long)((long)v14 + 1);
-        v44 = (long)v14;
-        if (1 <= (long)v14) {
-          v14 = (char *)sub_26dc6(v32,&v18,v14,0xffffffffffffffff,1);
-          v32 = v14;
-          dat_33780 = v14;
-        }
-        v14 = mempcpy(&v32[v21],optarg,v43);
-        v45 = v14;
-        *v14 = 10;
-        v14 = (long)(v43 + v21);
-        v14 = (long)sub_687b(v32,v21,(long)v14 + 1,0x290d2);
-        v21 = (long)v14;
+        v39 = strlen(optarg);
+        v40 = v39 + (v17 - v14) + 1;
+        if (1 <= v40) {
+          dat_33780 = (char *)sub_26dc6(v28,&v14,v40,0xffffffffffffffff,1);
+          v28 = dat_33780;
+        }
+        v41 = mempcpy(&v28[v17],optarg,v39);
+        *v41 = 10;
+        v17 = sub_687b(v28,v17,v39 + v17 + 1,0x290d2);
         break;
       case 0x66:
-        v13 = strcmp(optarg,"-");
-        if (v13) { // branch-flip
-          v14 = (dat_338b3) ? "rb" : "r"; // branch-flip
-          v14 = fopen(optarg,v14);
-          v7 = optarg;
-          v33 = v14;
-          if (!v14) {
-            v14 = (int *)__errno_location();
-            error(2,*(int *)v14,"%s",v7);
-          }
+        v9 = strcmp(optarg,"-");
+        if (v9) { // branch-flip
+          v11 = (dat_338b3) ? "rb" : "r"; // branch-flip
+          v29 = fopen(optarg,v11);
+          v11 = optarg;
+          if (!v29)
+            error(2,*__errno_location(),"%s",v11);
         }
         else {
           if (dat_338b3)
             sub_2716d(0,0);
-          v33 = stdin;
+          v29 = stdin;
         }
-        v34 = v21;
+        v30 = v17;
         while( true ) {
-          v14 = (long)(v34 - v18);
-          v14 = (long)((long)v14 + 2);
-          v41 = (long)v14;
-          if (1 <= (long)v14) {
-            v14 = (char *)sub_26dc6(v32,&v18,v14,0xffffffffffffffff,1);
-            v32 = v14;
-            dat_33780 = v14;
-          }
-          v14 = (long)(v18 - (v34 + 1));
-          v14 = (long)fread_unlocked(&v32[v34],1,v14,v33);
-          v42 = (long)v14;
-          if (!v14) break;
-          v34 += (long)v14;
-        }
-        v14 = (int *)__errno_location();
-        v28 = *(int *)v14;
-        v13 = ferror_unlocked(v33);
-        if (!v13) {
-          v28 = 0;
-          if (v33 != stdin) { // branch-flip
-            v13 = fclose(v33);
-            if (v13) {
-              v14 = (int *)__errno_location();
-              v28 = *(int *)v14;
-            }
+          v37 = (v30 - v14) + 2;
+          if (1 <= v37) {
+            dat_33780 = (char *)sub_26dc6(v28,&v14,v37,0xffffffffffffffff,1);
+            v28 = dat_33780;
+          }
+          v38 = fread_unlocked(&v28[v30],1,v14 - (v30 + 1),v29);
+          if (!v38) break;
+          v30 += v38;
+        }
+        v24 = *__errno_location();
+        v9 = ferror_unlocked(v29);
+        if (!v9) {
+          v24 = 0;
+          if (v29 != stdin) { // branch-flip
+            v9 = fclose(v29);
+            if (v9)
+              v24 = *__errno_location();
           }
           else {
-            clearerr_unlocked(v33);
+            clearerr_unlocked(v29);
           }
         }
-        if (v28)
-          error(2,v28,"%s",optarg);
-        if ((v34 != v21) && (v14 = &v32[v34 + -1], *v14 != '\n')) {
-          v15 = v34 + 1;
-          v14 = &v32[v34];
-          *v14 = '\n';
-          v34 = v15;
+        if (v24)
+          error(2,v24,"%s",optarg);
+        if ((v30 != v17) && (v28[v30 + -1] != '\n')) {
+          v12 = v30 + 1;
+          v28[v30] = '\n';
+          v30 = v12;
         }
-        v14 = (long)sub_687b(v32,v21,v34,optarg);
-        v21 = (long)v14;
+        v17 = sub_687b(v28,v17,v30,optarg);
         break;
       case 0x68:
-        v27 = 0xffffffff;
+        v23 = 0xffffffff;
         break;
       case 0x69:
       case 0x79:
@@ -3713,11 +3660,9 @@
         dat_33844 = 1;
         break;
       case 0x6d:
-        v13 = sub_27281(optarg,0,10,0x33870,0x290d2);
-        if (2 <= v13) {
-          v14 = gettext("invalid max count");
-          error(2,0,v14);
-        }
+        v7 = sub_27281(optarg,0,10,0x33870,0x290d2);
+        if (2 <= v7)
+          error(2,0,gettext("invalid max count"));
         break;
       case 0x6e:
         dat_33852 = 1;
@@ -3728,14 +3673,13 @@
       case 0x72:
 label_b073:
         dat_33010 = 3;
-        v26 = v1;
+        v22 = v9;
         break;
       case 0x73:
         dat_33755 = 1;
         break;
       case 0x75:
-        v14 = gettext("warning: --unix-byte-offsets (-u) is obsolete");
-        error(0,0,v14);
+        error(0,0,gettext("warning: --unix-byte-offsets (-u) is obsolete"));
         break;
       case 0x76:
         dat_33851 = '\x01';
@@ -3751,14 +3695,13 @@
         dat_33183 = 0;
         break;
       case 0x80:
-        v13 = strcmp(optarg,"binary");
-        if (v13) { // branch-flip
-          v13 = strcmp(optarg,"text");
-          if (v13) { // branch-flip
-            v13 = strcmp(optarg,"without-match");
-            if (v13) {
-              v14 = gettext("unknown binary-files type");
-              error(2,0,v14);
+        v9 = strcmp(optarg,"binary");
+        if (v9) { // branch-flip
+          v9 = strcmp(optarg,"text");
+          if (v9) { // branch-flip
+            v9 = strcmp(optarg,"without-match");
+            if (v9) {
+              error(2,0,gettext("unknown binary-files type"));
               goto label_b188;
             }
             dat_33840 = 2;
@@ -3774,12 +3717,12 @@
       case 0x81:
 label_b188:
         if (optarg) { // branch-flip
-          v13 = sub_11390(optarg,"always");
-          if (((v13) && (v13 = sub_11390(optarg,"yes"), v13)) && (v13 = sub_11390(optarg,"force"), v13)) {
-            v13 = sub_11390(optarg,"never");
-            if (((v13) && (v13 = sub_11390(optarg,"no"), v13)) && (v13 = sub_11390(optarg,"none"), v13)) {
-              v13 = sub_11390(optarg,"auto");
-              if (((v13) && (v13 = sub_11390(optarg,"tty"), v13)) && (v13 = sub_11390(optarg,"if-tty"), v13))
+          v9 = sub_11390(optarg,"always");
+          if (((v9) && (v9 = sub_11390(optarg,"yes"), v9)) && (v9 = sub_11390(optarg,"force"), v9)) {
+            v9 = sub_11390(optarg,"never");
+            if (((v9) && (v9 = sub_11390(optarg,"no"), v9)) && (v9 = sub_11390(optarg,"none"), v9)) {
+              v9 = sub_11390(optarg,"auto");
+              if (((v9) && (v9 = sub_11390(optarg,"tty"), v9)) && (v9 = sub_11390(optarg,"if-tty"), v9))
                 dat_33750 = 1;
               else {
                 dat_33758 = 2;
@@ -3799,55 +3742,31 @@
         break;
       case 0x82:
         sub_1c12b(optarg);
-        for (v31 = 0; v31 <= 1; v31 = v31 + 1) {
-          v14 = (long)(long)v31;
-          if (!*(long *)((long)v14 * 8 + 0x337b0)) {
-            v14 = (unsigned long)sub_1c384();
-            v10 = v14;
-            v14 = (long)(long)v31;
-            *(unsigned long *)((long)v14 * 8 + 0x337b0) = v10;
-          }
-          v13 = sub_7316(v31 != 0);
-          v8 = v13;
-          v14 = (long)(long)v31;
-          sub_1cca9(*(unsigned long *)((long)v14 * 8 + 0x337b0),optarg,v8);
+        for (v27 = 0; v27 <= 1; v27 = v27 + 1) {
+          if (!*(long *)((long)v27 * 8 + 0x337b0))
+            *(unsigned long *)((long)v27 * 8 + 0x337b0) = sub_1c384();
+          v8 = sub_7316(v27 != 0);
+          sub_1cca9(*(unsigned long *)((long)v27 * 8 + 0x337b0),optarg,v8);
         }
         break;
       case 0x83:
       case 0x86:
-        for (v29 = 0; v29 <= 1; v29 = v29 + 1) {
-          v14 = (long)(long)v29;
-          if (!*(long *)((long)v14 * 8 + 0x337a0)) {
-            v14 = (unsigned long)sub_1c384();
-            v10 = v14;
-            v14 = (long)(long)v29;
-            *(unsigned long *)((long)v14 * 8 + 0x337a0) = v10;
-          }
-          v16 = (v9 != 0x86) ? 0 : 0x20000000; // branch-flip
-          v13 = sub_7316(v29 != 0);
-          v16 |= v13;
-          v14 = (long)(long)v29;
-          sub_1cca9(*(unsigned long *)((long)v14 * 8 + 0x337a0),optarg,v16);
+        for (v25 = 0; v25 <= 1; v25 = v25 + 1) {
+          if (!*(long *)((long)v25 * 8 + 0x337a0))
+            *(unsigned long *)((long)v25 * 8 + 0x337a0) = sub_1c384();
+          v7 = (v10 != 0x86) ? 0 : 0x20000000; // branch-flip
+          sub_1cca9(*(unsigned long *)((long)v25 * 8 + 0x337a0),optarg,v7 | sub_7316(v25 != 0));
         }
         break;
       case 0x84:
-        for (v30 = 0; v30 <= 1; v30 = v30 + 1) {
-          v14 = (long)(long)v30;
-          if (!*(long *)((long)v14 * 8 + 0x337a0)) {
-            v14 = (unsigned long)sub_1c384();
-            v10 = v14;
-            v14 = (long)(long)v30;
-            *(unsigned long *)((long)v14 * 8 + 0x337a0) = v10;
-          }
-          v13 = sub_7316(v30 != 0);
-          v8 = v13;
-          v14 = (long)(long)v30;
-          v13 = sub_1d272(sub_1cca9,*(unsigned long *)((long)v14 * 8 + 0x337a0),optarg,v8,10);
-          v7 = optarg;
-          if (v13) {
-            v14 = (int *)__errno_location();
-            error(2,*(int *)v14,"%s",v7);
-          }
+        for (v26 = 0; v26 <= 1; v26 = v26 + 1) {
+          if (!*(long *)((long)v26 * 8 + 0x337a0))
+            *(unsigned long *)((long)v26 * 8 + 0x337a0) = sub_1c384();
+          v8 = sub_7316(v26 != 0);
+          v9 = sub_1d272(sub_1cca9,*(unsigned long *)((long)v26 * 8 + 0x337a0),optarg,v8,10);
+          v11 = optarg;
+          if (v9)
+            error(2,*__errno_location(),"%s",v11);
         }
         break;
       case 0x85:
@@ -3865,136 +3784,101 @@
     }
   } while( true );
 label_bb33:
-  if ((*v35 != ';') && (v13 = (int)*v35, v12 = sub_10eab(v13), !v12)) goto label_bb59;
-  v14 = &v35[1];
-  if (!*v14) {
-    dat_33030 = v39;
-    dat_33028 = v39;
+  if ((*v31 != ';') && (v6 = *v31, !sub_10eab((int)v6))) goto label_bb59;
+  if (!v31[1]) {
+    dat_33030 = v35;
+    dat_33028 = v35;
 label_bb59:
     sub_9ddf();
-    if ((v39 == dat_33028) || (v39 == dat_33030)) {
-      v14 = gettext("warning: GREP_COLOR=\'%s\' is deprecated; use GREP_COLORS=\'mt=%s\'");
-      error(0,0,v14,v39,v39);
-    }
+    if ((v35 == dat_33028) || (v35 == dat_33030))
+      error(0,0,gettext("warning: GREP_COLOR=\'%s\' is deprecated; use GREP_COLORS=\'mt=%s\'"),v35,v35);
 label_bbb8:
     sub_6f46();
-    if (v25 <= -1)
-      v25 = 0;
-    if (((v25 == 2) || (v25 == 1)) || (!v25)) {
+    if (v21 <= -1)
+      v21 = 0;
+    if (((v21 == 2) || (v21 == 1)) || (!v21)) {
       if (dat_33180)
         sub_a011();
-      if (v25 != 2) { // branch-flip
-        if (2 <= dat_33788) {
-          v13 = sub_a4d3(v25,v32,&v21);
-          v25 = v13;
-        }
+      if (v21 != 2) { // branch-flip
+        if (2 <= dat_33788)
+          v21 = sub_a4d3(v21,v28,&v17);
       }
       else {
         if (dat_331a0 != '\x01') { // branch-flip
           if ((dat_33788 != 1) || (!dat_33181))
-            v6 = 0;
+            v4 = 0;
           else {
-            v6 = 1;
+            v4 = 1;
           }
         }
         else {
-          v12 = sub_9f7b(v32,v21);
-          if ((v12) || ((dat_33180 && (v12 = sub_a29a(v32,v21), v12 != '\x01'))))
-            v6 = 1;
+          v6 = sub_9f7b(v28,v17);
+          if ((v6) || ((dat_33180 && (v6 = sub_a29a(v28,v17), v6 != '\x01'))))
+            v4 = 1;
           else {
-            v6 = 0;
+            v4 = 0;
           }
         }
-        if (v6) {
-          sub_a32f(0x33780,&v21);
-          v32 = dat_33780;
-          v25 = 0;
+        if (v4) {
+          sub_a32f(0x33780,&v17);
+          v28 = dat_33780;
+          v21 = 0;
         }
       }
     }
-    v14 = (long)(long)v25;
-    v14 = (long)((long)v14 * 0x20);
-    dat_337d8 = *(void **)((long)v14 + 0x327b8);
-    v14 = (long)(long)v25;
-    v14 = (long)((long)v14 * 0x20);
-    v3 = (unsigned long *)((long)v14 + 0x327b0);
-    v14 = (long)(long)v25;
-    v14 = (long)((long)v14 * 0x20);
-    v14 = (unsigned long)(*(void *)*v3)(v32,v21,(long)*(int *)((long)v14 + 0x327ac),dat_33758 || dat_3375c);
-    dat_337e0 = v14;
-    v65 = 0;
-    v12 = dat_33183;
-    v22 = v12;
-    v66 = 0;
-    v14 = &v22;
-    v14 = (long)(*dat_337d8)(dat_337e0,v14,1,v23,0);
-    dat_33831 = (v14 == NULL) == (bool)dat_33851;
-    v13 = a0 - optind;
-    v1 = v13;
-    if ((v27) || (2 <= (int)v13)) {
-      v13 = ~v27;
-      v13 >>= 0x1f;
-    }
+    dat_337d8 = *(void **)((long)v21 * 0x20 + 0x327b8);
+    dat_337e0 = (**(void **)((long)v21 * 0x20 + 0x327b0))(v28,v17,(long)*(int *)((long)v21 * 0x20 + 0x327ac),dat_33758 || dat_3375c);
+    v61 = 0;
+    v18 = dat_33183;
+    v62 = 0;
+    v12 = (*dat_337d8)(dat_337e0,&v18,1,v19,0);
+    dat_33831 = (v12 == 0) == (bool)dat_33851;
+    argc -= optind;
+    if ((v23) || (2 <= argc))
+      dat_33848 = ~v23 >> 0x1f;
     else {
-      v13 = (unsigned int)(dat_33010 == 3);
-      v13 = -v13;
+      dat_33848 = -(unsigned int)(dat_33010 == 3);
     }
-    dat_33848 = v13;
     if (dat_338b3)
       sub_2716d(1,0);
-    v14 = (unsigned long)sysconf(0x1e);
-    v11 = (unsigned long)v14;
-    v40 = (unsigned long)v14;
-    if ((1 <= (long)v14) && ((long)v14 <= 0x3ffffffffffffffb)) {
-      v4._8_8_ = 0;
-      v4._0_8_ = v14;
-      if (SUB168((ZEXT816(0) << 0x40 | ZEXT816(0x18000)) % v4._0_16_,0)) { // branch-flip
-        v5._8_8_ = 0;
-        v5._0_8_ = v14;
-        v14 = (long)((long)v14 - SUB168((ZEXT816(0) << 0x40 | ZEXT816(0x18000)) % v5._0_16_,0));
-        v15 = (long)v14 + 0x18000;
+    v36 = sysconf(0x1e);
+    if ((1 <= (long)v36) && ((long)v36 <= 0x3ffffffffffffffb)) {
+      v2._8_8_ = 0;
+      v2._0_8_ = v36;
+      if (SUB168((ZEXT816(0) << 0x40 | ZEXT816(0x18000)) % v2._0_16_,0)) { // branch-flip
+        v3._8_8_ = 0;
+        v3._0_8_ = v36;
+        v12 = (v36 - SUB168((ZEXT816(0) << 0x40 | ZEXT816(0x18000)) % v3._0_16_,0)) + 0x18000;
       }
       else {
-        v15 = 0x18000;
+        v12 = 0x18000;
       }
-      v14 = (long)(v11 + v15);
-      v14 = (long)((long)v14 + 8);
-      dat_337f8 = (long)v14;
-      dat_33818 = v11;
-      v14 = (unsigned long)sub_26b10(v14);
-      dat_337f0 = v14;
-      v13 = dat_33014 & 2;
-      if ((v13) && (!dat_337cc))
+      dat_337f8 = v36 + v12 + 8;
+      dat_33818 = v36;
+      dat_337f0 = sub_26b10(dat_337f8);
+      if ((dat_33014 & 2) && (!dat_337cc))
         dat_337cc = 1;
-      if (1 <= v1) { // branch-flip
-        v14 = (long)(long)optind;
-        v14 = (long *)(a1 + (long)v14 * 8);
-        v36 = (long *)v14;
-      }
-      else if ((dat_33010 != 3) || (v26 <= 0))
-        v36 = (long *)0x32730;
+      if (1 <= argc) // branch-flip
+        v32 = &argv[optind];
+      else if ((dat_33010 != 3) || (v22 <= 0))
+        v32 = (char **)0x32730;
       else {
-        v36 = (long *)0x32720;
+        v32 = (char **)0x32720;
         dat_337c8 = 1;
       }
-      v24 = 1;
+      v20 = 1;
       do {
-        v2 = &v36[1];
-        v15 = *v36;
-        v36 = v2;
-        v12 = sub_992b(v15);
-        v24 = (v12 & v24) != 0;
-      } while (*v36);
-      if (!dat_337c9) {
-        v12 = v24;
-        return v12;
-      }
-      v12 = 2;
-      return v12;
+        v1 = &v32[1];
+        v11 = *v32;
+        v32 = v1;
+        v20 = (sub_992b(v11) & v20) != 0;
+      } while (*v32);
+      v7 = (dat_337c9) ? 2 : (unsigned int)v20; // branch-flip
+      return v7;
     }
     abort(); // no-return
   }
-  v35 = &v35[1];
+  v31 = &v31[1];
   goto label_bb33;
 }
 
