--- /home/mahaloz/kwt/elfmain/.scratch/realsweep/off/O2_findutils_find.c	2026-09-16 20:49:37.698010878 +0000
+++ /home/mahaloz/kwt/elfmain/.scratch/realsweep/on/O2_findutils_find.c	2026-09-16 20:49:26.837989510 +0000
@@ -1173,19 +1173,19 @@
   return v1;
 }
 
-// Function: sub_7240 @ 0x7240
-unsigned long sub_7240(int a0,long *a1)
+// Function: main @ 0x7240
+int main(int argc,char **argv,char **envp)
 {
   int v1; // eax
-  unsigned long v2; // rax
-  char *v3;
+  int v2; // eax
+  int v3; // eax
   unsigned long v4; // rax
-  unsigned long v5; // rax
+  char *v5;
   
-  v3 = (char *)*a1;
-  if (!v3)
-    v3 = "find";
-  sub_21e40(v3);
+  v5 = *argv;
+  if (!v5)
+    v5 = "find";
+  sub_21e40(v5);
   sub_da20();
   dat_38618 = 0;
   dat_38604 = 0;
@@ -1200,25 +1200,25 @@
     bindtextdomain("findutils","/usr/local/share/locale");
     textdomain("findutils");
     if (!sub_27a00(sub_17a40)) {
-      v1 = sub_e180(a0,a1);
+      v1 = sub_e180(argc,argv);
       if (dat_38660 & 2)
         dat_38670 = sub_dec0;
       if (dat_38660 & 0x80)
         __fprintf_chk(stderr,1,"cur_day_start = %s",ctime(0x38648));
-      v2 = sub_c9f0(a0,a1,v1);
-      if (sub_8070(a0 - v1,&a1[v1])) {
-        sub_9e00(v2);
+      v4 = sub_c9f0(argc,argv,v1);
+      if (sub_8070(argc - v1,&argv[v1])) {
+        sub_9e00(v4);
         sub_dab0();
       }
-      return (unsigned long)dat_38604;
+      return dat_38604;
     }
-    v3 = dcgettext(NULL,"The atexit library function failed",5);
-    error(1,*__errno_location(),v3);
-    return v4;
+    v5 = dcgettext(NULL,"The atexit library function failed",5);
+    error(1,*__errno_location(),v5);
+    return v2;
   }
-  v3 = dcgettext(NULL,"Failed to initialize shared-file hash table",5);
-  error(1,*__errno_location(),v3);
-  return v5;
+  v5 = dcgettext(NULL,"Failed to initialize shared-file hash table",5);
+  error(1,*__errno_location(),v5);
+  return v3;
 }
 
 // Function: sub_7420 @ 0x7420
@@ -1227,7 +1227,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_37fe0)(sub_7240,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_37fe0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
