--- /home/mahaloz/kwt/elfmain/.scratch/sweep/off/tailcallframe_x86_64.c	2026-09-16 19:25:05.485163342 +0000
+++ /home/mahaloz/kwt/elfmain/.scratch/sweep/on/tailcallframe_x86_64.c	2026-09-16 19:25:05.337162959 +0000
@@ -27,8 +27,8 @@
   (*dat_3ff8)(); // jump-as-call
 }
 
-// Function: sub_1050 @ 0x1050
-int sub_1050(void)
+// Function: main @ 0x1050
+int main(int argc,char **argv)
 {
   (*(void *)*dat_4010)(0x4040,3);
   return dat_4048 + dat_4044 + dat_404c;
@@ -40,7 +40,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_3fd8)(sub_1050,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_3fd8)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
