--- armlibcmain_le32 (option elfmain off)
+++ armlibcmain_le32 (default)
@@ -47,27 +47,29 @@
 }
 
 // Function: main @ 0x103dc
-void main(unsigned int a0,unsigned int a1,unsigned int a2)
+int main(int argc,char **argv,char **envp)
 {
   char *v1; // r0
   unsigned int v2;
-  char v3 [32]; // stack - 0x2c
-  void *v4; // lr
-  int v5; // stack - 0xc
+  int v3; // r0
+  char v4 [32]; // stack - 0x2c
+  void *v5; // lr
+  int v6; // stack - 0xc
   
-  v5 = __stack_chk_guard;
-  __printf_chk(2,"kuna armlibcmain prompt",a2,0);
-  v1 = fgets(v3,0x20,stdin);
+  v6 = __stack_chk_guard;
+  __printf_chk(2,"kuna armlibcmain prompt",envp,0);
+  v1 = fgets(v4,0x20,stdin);
   if (v1) { // branch-flip
     v2 = 0;
-    if (v3[0] != 'k')
+    if (v4[0] != 'k')
       v2 = 1;
   }
   else {
     v2 = 1;
   }
-  if (v5 != __stack_chk_guard)
+  if (v6 != __stack_chk_guard)
     __stack_chk_fail(); // no-return
-  (*v4)(v2); // jump-as-call
+  v3 = (*v5)(v2); // jump-as-call
+  return v3;
 }
 
