--- emptystrconst_x86_64 (option elfmain off)
+++ emptystrconst_x86_64 (default)
@@ -27,7 +27,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_403ff0)(sub_401189,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_403ff0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
@@ -71,10 +71,10 @@
   return v1 + v2 + strlen("Hi");
 }
 
-// Function: sub_401189 @ 0x401189
-void sub_401189(void)
+// Function: main @ 0x401189
+int main(int argc,char **argv)
 {
-  sub_401136();
+  return sub_401136();
 }
 
 // Function: _DT_FINI @ 0x401198
