--- O0/coreutils/stripped/ls (option elfmain off)
+++ O0/coreutils/stripped/ls (default)
@@ -915,7 +915,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_2afa0)(sub_6882,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_2afa0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
@@ -1669,8 +1669,8 @@
   sub_6593(0);
 }
 
-// Function: sub_6882 @ 0x6882
-unsigned int sub_6882(int a0,unsigned long *a1) // warn: iteboolean: re-rolled 2 0/1 select diamond(s) to a boolean assignment in sub_6882
+// Function: main @ 0x6882
+int main(int argc,char **argv,char **envp) // warn: iteboolean: re-rolled 2 0/1 select diamond(s) to a boolean assignment in main
 {
   int v1;
   int v10; // stack - 0x44
@@ -1685,7 +1685,7 @@
   unsigned long v8; // stack - 0x28
   char v9 [16];
   
-  sub_18e89(*a1);
+  sub_18e89(*argv);
   setlocale(6,"");
   bindtextdomain("coreutils","/usr/local/share/locale");
   textdomain("coreutils");
@@ -1696,7 +1696,7 @@
   dat_2b368 = NULL;
   dat_2b370 = 0x8000000000000000;
   dat_2b378 = 0xffffffffffffffff;
-  v5 = sub_70af(a0,a1);
+  v5 = sub_70af(argc,argv);
   if (dat_2b3d4)
     sub_88ab();
   if (dat_2b3d4)
@@ -1736,12 +1736,12 @@
   dat_2b338 = 0;
   sub_96f3();
   v10 = v5; // branch-flip
-  if (1 <= a0 - v5) {
+  if (1 <= argc - v5) {
     do {
       v1 = v10 + 1;
-      sub_9a40(a1[v10],0,0,1,0x1f27b);
+      sub_9a40(argv[v10],0,0,1,0x1f27b);
       v10 = v1;
-    } while (v1 < a0);
+    } while (v1 < argc);
   }
   else if (dat_2b3f1) // branch-flip
     sub_9a40(".",3,0,1,0x1f27b);
@@ -1755,7 +1755,7 @@
     if (dat_2b368)
       sub_5422(10);
   }
-  else if (((a0 - v5 <= 1) && (dat_2b368)) && (!dat_2b368[3]))
+  else if (((argc - v5 <= 1) && (dat_2b368)) && (!dat_2b368[3]))
     dat_2b428 = 0;
   while (v4 = dat_2b368, dat_2b368) {
     if ((dat_2b320) && (!*dat_2b368)) {
