--- O2/gzip/stripped/gzip (option elfmain off)
+++ O2/gzip/stripped/gzip (default)
@@ -645,8 +645,8 @@
   return v1;
 }
 
-// Function: sub_39a0 @ 0x39a0
-void sub_39a0(int a0,char **a1)
+// Function: main @ 0x39a0
+int main(int argc,char **argv,char **envp)
 {
   unsigned int *v1;
   char **v10; // stack - 0xe0
@@ -672,14 +672,14 @@
   long v9; // rcx
   
   v22 = *(unsigned long *)(v20 + 0x28);
-  v7 = (char *)sub_d290(*a1);
+  v7 = (char *)sub_d290(*argv);
   dat_1a850 = v7;
   v8 = strlen(v7);
   if (5 <= v8) {
     if (!strcmp(&v7[v8 - 4],".exe"))
       v7[v8 - 4] = '\0';
   }
-  v10 = a1;
+  v10 = argv;
   dat_dc9d0 = sub_d2b0(&v11,&v10,"GZIP");
   v17 = NULL;
   if (dat_dc9d0)
@@ -712,7 +712,7 @@
   }
   else {
 label_3b65:
-    v4 = getopt_long(a0,a1,"ab:cdfhH?klLmMnNqrS:tvVZ123456789",(void *)0x177c0,&v12);
+    v4 = getopt_long(argc,argv,"ab:cdfhH?klLmMnNqrS:tvVZ123456789",(void *)0x177c0,&v12);
     v3 = optind;
     if (v4 < 0) {
       if (dat_18028 <= -1)
@@ -772,9 +772,9 @@
         }
         v19 += 1;
       } while (v19 != 6);
-      if (a0 != v3) { // branch-flip
-        while (optind < a0) {
-          v7 = a1[optind];
+      if (argc != v3) { // branch-flip
+        while (optind < argc) {
+          v7 = argv[optind];
           optind += 1;
           v6 = strcmp(v7,"-");
           v4 = dat_1a860;
@@ -795,7 +795,7 @@
         sub_d4e0(); // no-return
       }
       if (dat_dc9e8) {
-        if ((!dat_1a85c) && (2 <= a0 - v3))
+        if ((!dat_1a85c) && (2 <= argc - v3))
           sub_65d0(0xffffffff);
         v4 = sub_e3e0(stdout);
         if (v4) {
@@ -809,8 +809,8 @@
     v5 = v4 - 0x31;
 label_3b44:
     if (v5 <= 0xd3) {
-      (*(void *)((long)*(int *)((unsigned long)v5 * 4 + 0x12f80) + 0x12f80))(); // jump-as-call
-      return;
+      v4 = *(int *)((unsigned long)v5 * 4 + 0x12f80);
+      return (*(void *)((long)v4 + 0x12f80))(); // jump-as-call
     }
     if ((v4 <= 0x82) || (v4 == 0xc2)) {
       sub_69d0(); // no-return, return-dupe
@@ -832,7 +832,7 @@
   char v1 [8];
   unsigned long v2; // stack + 0x0
   
-  (*dat_17fc0)(sub_39a0,v2,&Stack0000000000000008,0,0,a2,v1);
+  (*dat_17fc0)(main,v2,&Stack0000000000000008,0,0,a2,v1);
   do {
   } while( true );
 }
